CMakeLists.txt

```diff
@@ -183,6 +183,12 @@ if (BUILD_TESTING AND BUILDSYSTEM_TESTING)
   add_subdirectory(examples)
 endif()
 
+# Codecheck
+set(CPPCHECK_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/include)
+set(IGNITION_CMAKE_CODECHECK_DIR "${CMAKE_CURRENT_SOURCE_DIR}/codecheck")
+include(IgnCodeCheck)
+ign_setup_target_for_codecheck()
```

Comment:
It looks like GitHub actions isn't happy with this line, but I'm not sure why:
```
  CMake Error at cmake/IgnCodeCheck.cmake:35 (add_custom_target):
  -- Adding codecheck target
    add_custom_target cannot create target "cppcheck" because another target
    with the same name already exists.  The existing target is a custom target
    created in source directory "/github/workspace/examples".  See
    documentation for policy CMP0002 for more details.
  Call Stack (most recent call first):
    CMakeLists.txt:190 (ign_setup_target_for_codecheck)
```

Reply:
(The call to `cmake ..` in GitHub actions is currently failing)

Reply:
I introduced the `cppcheck` target in a previous PR: https://github.com/ignitionrobotics/ign-cmake/pull/117

This introduced a `cpplint` and `cppcheck` target, which are both executed when you do `make codecheck`

Reply:
Interesting, it worked for me locally. Let me investigate more

Reply:
Ok, I have a fix, pushing soon

Reply:
That `codecheck` target was only being added with `-DBUILDSYSTEM_TESTING=True`. This should fix it, and I also fixed codecheck errors on the examples folder:

9b258c587c46a179e095c7f21dd03e14ce6e0c4f